File: PrimaMateria/PrimaMateria/Classes/XTRAppDelegate.m
//
//  XTRAppDelegate.m
//  PrimaMateria
//
//  Created by Jerry Porter on 11/21/11.
//  Copyright 2016 xTrensa. All rights reserved.
//

@implementation XTRAppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
    self.window.rootViewController.modalPresentationStyle = UIModalPresentationCurrentContext;
    
    [XTRAppDelegate applyStyle];
    return YES;
}

+(UIStoryboard *)storyboard {
    return [UIStoryboard storyboardWithName: MAIN_STORY_BOARD bundle:nil];
}

+ (void)applyStyle {
    UINavigationBar *navigationBarAppearance = [UINavigationBar appearance];
    UIToolbar *toolBarAppearance = [UIToolbar appearance];
    UITabBar *tabBarAppearance = [UITabBar appearance];
    UITabBarItem *tabBarItemAppearance = [UITabBarItem appearance];
    UISegmentedControl *segmentedControlBarAppearance = [UISegmentedControl appearance];
    UISwitch *switchAppearance = [UISwitch appearance];

    NSDictionary *navigationBarTextAttributes = @{
                                                  NSForegroundColorAttributeName: UIColor.whiteColor,
                                                  NSFontAttributeName: [UIFont boldSystemFontOfSize: 20]
                                                  };
    NSDictionary *tabBarNormalItemTextAttributes = @{
                                                  NSForegroundColorAttributeName: UIColor.whiteColor,
                                                  NSFontAttributeName: [UIFont boldSystemFontOfSize: 14]
                                                  };
    NSDictionary *tabBarSelectedItemTextAttributes = @{
                                                     NSForegroundColorAttributeName: [UIColor colorWithHexString:@"333333"],
                                                     NSFontAttributeName: [UIFont boldSystemFontOfSize: 14]
                                                     };

    NSDictionary *segmentedControlTextAttributes = @{
                                                     NSForegroundColorAttributeName: UIColor.whiteColor,
                                                     NSFontAttributeName: [UIFont systemFontOfSize: 13]
                                                     };
    
    navigationBarAppearance.titleTextAttributes = navigationBarTextAttributes;
    navigationBarAppearance.tintColor = [UIColor colorWithHexString:@"eeeeee"];
    
    toolBarAppearance.tintColor = [UIColor colorWithHexString:@"eeeeee"];

    [tabBarItemAppearance setTitleTextAttributes:tabBarNormalItemTextAttributes forState:UIControlStateNormal];
    [tabBarItemAppearance setTitleTextAttributes:tabBarSelectedItemTextAttributes forState:UIControlStateSelected];

    tabBarAppearance.tintColor = [UIColor colorWithHexString:@"333333"];
    tabBarAppearance.backgroundColor = [UIColor colorWithHexString:@"999999"];
    //[tabBarAppearance setSelectionIndicatorImage: [UIImage imageNamed: @"SelectedTab"]];
    

    [segmentedControlBarAppearance setTitleTextAttributes: segmentedControlTextAttributes forState: UIControlStateNormal];
    
    switchAppearance.onTintColor = [UIColor colorWithHexString:@"666666"];
}

-(UIInterfaceOrientationMask)application:(UIApplication *)application supportedInterfaceOrientationsForWindow:(UIWindow *)window {
    return UIInterfaceOrientationMaskLandscape;
}

@end

File: PrimaMateria/PrimaMateria/Classes/XTRAppDelegate.swift
//
//  XTRAppDelegate.swift
//  PrimaMateria
//
//  Created by Jerry Porter on 4/20/16.
//  Copyright ©2018 xTrensa. All rights reserved.
//

class XTRAppDelegate : UIResponder, UIApplicationDelegate {
    
    var window : UIWindow?
    
    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
        XTRAppDelegate.applyStyle()
        //        let fontFamiles : [Any] = UIFont.familyNames()
        //        print("%@", fontFamiles)
        //        let fontNames : [Any] = UIFont.fontNamesForFamilyName("Verdana")
        //        print("%@", fontNames)
        return true
    }
    
    class func storyboard() -> UIStoryboard {
        return UIStoryboard(name: MAIN_STORY_BOARD, bundle: nil)
    }
    
    class func applyStyle() {
        let navigationBarAppearance = UINavigationBar.appearance()
        let toolBarAppearance = UIToolbar.appearance()
        let tabBarAppearance = UITabBar.appearance()
        let tabBarItemAppearance = UITabBarItem.appearance()
        let segmentedControlBarAppearance = UISegmentedControl.appearance()
        let switchAppearance = UISwitch.appearance()
        
        let navigationBarTextAttributes: [NSAttributedStringKey: AnyObject]? = [
            NSAttributedStringKey.foregroundColor: UIColor.white,
            NSAttributedStringKey.font: UIFont.boldSystemFont(ofSize: 20)
        ]
        
        let segmentedControlTextAttributes: [NSAttributedStringKey: AnyObject]?
        let tabBarNormalItemTextAttributes: [NSAttributedStringKey: AnyObject]?
        let tabBarSelectedItemTextAttributes: [NSAttributedStringKey: AnyObject]?
        
        if #available(iOS 11.0, *) {
            segmentedControlTextAttributes = [
                NSAttributedStringKey.foregroundColor: UIColor.color(hexString: "4093ff"),
                NSAttributedStringKey.font: UIFont.boldSystemFont(ofSize: 13)
            ]
            tabBarNormalItemTextAttributes = [
                NSAttributedStringKey.foregroundColor: UIColor.white,
                NSAttributedStringKey.font: UIFont.boldSystemFont(ofSize: 14)
            ]
            tabBarSelectedItemTextAttributes = [
                NSAttributedStringKey.foregroundColor: UIColor.color(hexString: "4093ff"),
                NSAttributedStringKey.font: UIFont.boldSystemFont(ofSize: 14)
            ]
            tabBarAppearance.tintColor = UIColor.color(hexString: "4093ff")
        } else {
            segmentedControlTextAttributes = [
                NSAttributedStringKey.foregroundColor: UIColor.color(hexString: "666666"),
                NSAttributedStringKey.font: UIFont.boldSystemFont(ofSize: 13)
            ]
            tabBarNormalItemTextAttributes = [
                NSAttributedStringKey.foregroundColor: UIColor.white,
                NSAttributedStringKey.font: UIFont.boldSystemFont(ofSize: 12)
            ]
            tabBarSelectedItemTextAttributes = [
                NSAttributedStringKey.foregroundColor: UIColor.color(hexString: "333333"),
                NSAttributedStringKey.font: UIFont.boldSystemFont(ofSize: 12)
            ]
            tabBarAppearance.tintColor = UIColor.color(hexString: "333333")
        }
        
        tabBarItemAppearance.setTitleTextAttributes(tabBarNormalItemTextAttributes, for: UIControlState())
        tabBarItemAppearance.setTitleTextAttributes(tabBarSelectedItemTextAttributes, for: .selected)
        
        tabBarAppearance.backgroundColor = UIColor.color(hexString: "999999")
        
        segmentedControlBarAppearance.setTitleTextAttributes(segmentedControlTextAttributes, for: .selected)
        
        navigationBarAppearance.titleTextAttributes = navigationBarTextAttributes
        navigationBarAppearance.tintColor = UIColor.color(hexString: "eeeeee")
        
        toolBarAppearance.tintColor = UIColor.color(hexString: "eeeeee")
        
        switchAppearance.onTintColor = UIColor.color(hexString: "666666")
    }
    
    func application(_ application: UIApplication, supportedInterfaceOrientationsFor window: UIWindow?) -> UIInterfaceOrientationMask {
        return .landscape
    }
    
}
